src/profile-logic/profile-data.js

```diff
@@ -150,6 +150,186 @@ export function getSampleCallNodes(
   });
 }
 
+// This function assumes that the path isn't empty
+export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
+  return path[path.length - 1];
+}
+
+// Returns the JS implementation information for a specific stack
+export function getJsImplementationForStack(
+  stackIndex: IndexIntoStackTable,
+  { stackTable, frameTable, stringTable }: Thread
+): string {
+  const frameIndex = stackTable.frame[stackIndex];
+  const jsImplementationStrIndex = frameTable.implementation[frameIndex];
+  const jsImplementation =
+    jsImplementationStrIndex === null
+      ? 'interpreter'
+      : stringTable.getString(jsImplementationStrIndex);
+
+  return jsImplementation;
+}
+
+type Breakdown = { [key: string]: Milliseconds } | null;
```

Comment:
I initially envisioned the same type Breakdown to be reused for the category breakdown too. I also thought that it can be used the same way, with the `Breakdown` component that you can see in https://github.com/devtools-html/perf.html/pull/1041/files#diff-9b0d8575e8c83d30afef14e67ce2f32fR49. So that's why I created a quite generic type.

Now maybe I can make it more strict here, but still have the `Breakdown` component taking the `Breakdown` type, because after all a stricter type should be accepted.

Reply:
> Also I think adding the | null to the ItemTimings definitions make it more obvious that the object may be null.

Got it, good feedback indeed, I'll do it !

Reply:
> type JsImplementation = 'interpreter' | 'baseline' | 'ion';

TBH I took some shortcut in the next patch; but I think you're right so I'll try to change it